src/ir/auth_logic/lowering_ast_datalog.cc

```diff
@@ -269,40 +263,39 @@ std::vector<datalog::DLIRAssertion> LoweringToDatalogPass::SaysAssertionToDLIR(
   return ret;
 }
 
-std::vector<datalog::DLIRAssertion> LoweringToDatalogPass::SaysAssertionsToDLIR(
+std::vector<datalog::Rule> LoweringToDatalogPass::SaysAssertionsToDLIR(
     const std::vector<SaysAssertion>& says_assertions) {
-  std::vector<datalog::DLIRAssertion> ret = {};
+  std::vector<datalog::Rule> ret = {};
   for (const SaysAssertion& says_assertion : says_assertions) {
     absl::c_move(SaysAssertionToDLIR(says_assertion), std::back_inserter(ret));
   }
   return ret;
 }
 
-std::vector<datalog::DLIRAssertion> LoweringToDatalogPass::QueriesToDLIR(
+std::vector<datalog::Rule> LoweringToDatalogPass::QueriesToDLIR(
     const std::vector<Query>& queries) {
-  return utils::MapIter<datalog::DLIRAssertion>(
+  return utils::MapIter<datalog::Rule>(
       queries, [this](const Query& query) {
         auto [main_pred, not_used] =
             FactToDLIR(query.principal(), query.fact());
         main_pred = PushPrincipal("says_", query.principal(), main_pred);
         datalog::Predicate lhs(query.name(), {"dummy_var"}, datalog::kPositive);
-        return datalog::DLIRAssertion(
-            datalog::DLIRCondAssertion(lhs, {main_pred, kDummyPredicate}));
+        return datalog::Rule(lhs, {main_pred, kDummyPredicate});
       });
 }
 
-datalog::DLIRProgram LoweringToDatalogPass::ProgToDLIR(const Program& program) {
+datalog::Program LoweringToDatalogPass::ProgToDLIR(const Program& program) {
   auto dlir_assertions = SaysAssertionsToDLIR(program.says_assertions());
   // We need to add a fact that says the dummy variable used in queries is
   // grounded.
-  datalog::DLIRAssertion dummy_assertion(kDummyPredicate);
+  datalog::Rule dummy_assertion(kDummyPredicate);
   absl::c_move(QueriesToDLIR(program.queries()),
                std::back_inserter(dlir_assertions));
   dlir_assertions.push_back(dummy_assertion);
 
   auto outputs = utils::MapIter<std::string>(
       program.queries(), [](const Query& query) { return query.name(); });
-  return datalog::DLIRProgram(dlir_assertions, outputs);
+  return datalog::Program(dlir_assertions, outputs);
```

Comment:
Should probably just inline the initializer of `outputs`.

Reply:
I think using output variable is better in terms of readability.